comp/rdnsquerier/impl/rdnsquerier_test.go

```diff
@@ -986,3 +993,81 @@ func TestCachePersist(t *testing.T) {
 	})
 	ts.validateExpected(t, expectedTelemetry)
 }
+
+func TestGetHostnameSync(t *testing.T) {
+	overrides := map[string]interface{}{
+		"network_devices.netflow.reverse_dns_enrichment_enabled": true,
+	}
+
+	ts := testSetup(t, overrides, true, nil, 0)
+	internalRDNSQuerier := ts.rdnsQuerier.(*rdnsQuerierImpl)
+
+	// Vic to reformat this test
+	// tests := []struct {
+	// 	description string
+	// 	ip          string
+	// 	expected    string
+	// 	errMsg      string
+	// }{
+	// 	{},
+	// }
+
+	ctx, cancel := context.WithTimeout(ts.ctx, 100*time.Second)
+	var hostname string
+	var err error
+
+	// // Test with invalid IP address
+	hostname, err = internalRDNSQuerier.GetHostnameSync(ctx, "invalid_ip")
+	assert.Error(t, err)
```

Comment:
Since we have multiple error scenarios, I think it would be good to check the errors themselves to see if they're the errors we expect. That way we know the code is returning an error where we expect.

```suggestion
    assert.Equal(t, "", hostname)
	require.Error(t, err)
	assert.Contains(t, err.Error(), "some error message")
```

By using `require` above you ensure that if the error is `nil` the test execution stops here and so we don't try the dereference in the next line.